app/components/ui/molecules/PeoplePicker.js

```diff
@@ -145,11 +147,57 @@ class PeoplePicker extends React.Component {
     this.setState({ searchValue })
   }
 
-  filterPeople(people) {
-    const searchValue = this.state.searchValue.toLowerCase()
-    return people.filter(person =>
-      person.name.toLowerCase().includes(searchValue),
-    )
+  filterPeople = (people, searchValue, field) => {
+    if (!searchValue) return people
+
+    const inputValue = searchValue.trim().toLowerCase()
+    if (!inputValue) return people
+
+    if (inputValue.split(' ').length > 1) {
+      const matches = people.filter(person =>
+        person[field].toLowerCase().includes(inputValue),
+      )
+      return matches.map(elem => {
+        const clonedElem = cloneDeep(elem)
+        clonedElem.matchIndex = elem[field].toLowerCase().indexOf(inputValue)
+        return clonedElem
+      })
+    }
+    return this.filterPeopleSingle(people, inputValue, field)
+  }
+
+  filterPeopleSingle = (people, inputValue, field) => {
+    let maxWords = 0
+    people.forEach(person => {
+      const numWords = person[field].match(/[^ ]+/g).length
+      if (numWords > maxWords) {
+        maxWords = numWords
+      }
+    })
+
+    let matches = []
+    const re = new RegExp(`^${escapeRegExp(inputValue)}`)
+    const filterNthMatches = (person, n) => {
+      const words = person[field].match(/[^ ]+/g)
+      if (words.length <= n) {
+        return false
+      }
+      return re.test(words[n].toLowerCase())
+    }
+
+    for (let i = 0; i < maxWords; i += 1) {
+      /* matches for ith word, starting at 0 */
+      const ithMatches = people.filter(person => filterNthMatches(person, i))
+      matches = matches.concat(ithMatches)
+    }
+    return matches
```

Comment:
I don't really understand what any of this is doing.

Reply:
We discussed this irl, I match after the first word, then after the second and so on.